zio-http/src/main/scala/zio/http/Driver.scala

```diff
@@ -0,0 +1,32 @@
+package zio.http
+
+import zio._
+
+import java.util.concurrent.atomic.AtomicReference
+
+trait Driver {
```

Comment:
I agree with @vigoo that rather than expose these atomic references in this callback-style API, we should refactor a bit.

Fundamentally, Driver does not need to expose the atomic references, only expose the capabilities. e.g.:

```scala
def install(httpApp: HttpApp[Any, Throwable])(implicit unsafe: Unsafe): Unit
def setErrorCallback(errorCallback: Server.ErrorCallback)(implicit unsafe: Unsafe): Unit
```

where the first one will add it to the atomic reference inside the Netty implementation, and the second one will do whatever it does.

AtomicReference feels like an implementation detail and we don't need to know about it.

This will let you delete `Context` entirely and stop threading these through layers, etc.

Reply:
OK, I _think_ this more in the spirit of what y'all are asking for, https://github.com/zio/zio-http/pull/1485/commits/9951ead005496a8b8ab7ca167a725187b0bfb07d.

Thoughts/concernes?